src/main/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannel.java

```diff
@@ -269,7 +315,11 @@ public void insertRecordToBuffer(SinkRecord kafkaSinkRecord) {
       // If we found reaching buffer size threshold or count based threshold, we will immediately
       // flush (Insert them)
       if (copiedStreamingBuffer != null) {
-        insertBufferedRecords(copiedStreamingBuffer);
+        if (this.enableSchematization && this.enableSchemaEvolution) {
+          insertBufferedRecordsWithRetry(copiedStreamingBuffer);
```

Comment:
do you think these two functions could be combined? If the only difference is the number of reties, we could pass a retry count of 0 to `insertBufferedRecords` and combine these two?

Reply:
There are a few differences that come to my mind:
1. The `insertBufferedRecords` will add rows with any types of error to DLQ while `insertBufferedRecordsWithRetry` will add rows that cannot or failed to be reinserted to DLQ.
2. The `insertBufferedRecords` will only reopen the channel when an SFException is thrown while the `insertBufferedRecordsWithRetry` will reopen the channel with each retry, which also leads to a difference that the the channel offset needs to be reset for the latter.

I see that there are now many functions look like twins, but I think combining them will make the code quite messy.